### PR TITLE
ENS Snapshot indexer

### PR DESCRIPTION
**Pull request by stepandel**
CleanShot 2024-03-20 at 21 17 11@2x

---

## PR-Codex overview
The focus of this PR is to add Snapshot proposals functionality and update related components.

### Detailed summary
- Added Snapshot proposal support with new components and utility functions
- Updated styles and imports in components
- Updated Prisma schema to include Snapshot proposals
- Added Snapshot route for fetching and upserting proposals into the database

> The following files were skipped due to too many changes: `src/lib/proposalUtils.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by stepandel** [comment]
This PR requires database update

File: src/components/Proposals/Proposal/Proposal.jsx [hunk 1]
```diff
@@ -7,26 +7,50 @@ import OPApprovalProposalStatus from "./OPApprovalProposalStatus";
...
+              <p>Snapshot Proposal</p>
+              <ArrowTopRightOnSquareIcon className="w-3 h-3 mt-1" />
+            </HStack>
+          ) : (
+            <HStack className={styles.cell_content_title} gap={1}>
+              {/* Warning: this assumes OP FND is the only proposer. Will need to maintain an array of OP Foundation proposals eventually */}
+              <div>
+                {proposalText}{" "}
+                <span className={styles.invisible_on_mobile}>
+                  {Tenant.current().namespace === "optimism" ? (
```
Comment (andreitr) [inline src/components/Proposals/Proposal/Proposal.jsx:40]: My preference is to start using constants for namespace flags. Something like this 
`{Tenant.current().namespace === TENANT_NAMESPACES.OPTIMISM}`

File: src/components/Proposals/Proposal/Proposal.jsx [hunk 2]
```diff
@@ -7,26 +7,50 @@ import OPApprovalProposalStatus from "./OPApprovalProposalStatus";
...
+                    "by The Optimism Foundation"
```
Comment (andreitr) [inline src/components/Proposals/Proposal/Proposal.jsx:41]: Good catch. I need to move this into the tenant.

File: src/lib/utils.ts [hunk 3]
```diff
@@ -21,16 +22,19 @@ export function bpsToString(bps: number) {
...
-  switch (proposalType) {
-    case "OPTIMISTIC":
-      return "Optimistic Proposal";
-    case "STANDARD":
-      return "Standard Proposal";
-    case "APPROVAL":
-      return "Approval Vote Proposal";
-    default:
-      return "Proposal";
+  if (Tenant.current().namespace === "optimism") {
```
Comment (andreitr) [inline src/lib/utils.ts:25]: Let's use a constant here as well.

**Review by andreitr — APPROVED** [review]
Left a few non-blocking comments. LGTM

**Pull request merged**